### PR TITLE
Temporaly removed Reporting and added Sync Gateway(template)

### PR DESCRIPTION
**Pull request by Anh3h**

### CODE REVIEW

**Comment by awasum** [comment]
Please...l have refactored template...so in Demo server...U need to use different Manager: SyncManager interface..

**Comment by awasum** [comment]
Also i still dont know why we need an API gateway for Sync..Maybe we will figure out how to remove that seamlessly later...or do u see a use case?

File: build.gradle [hunk 1]
```diff
@@ -32,6 +32,7 @@ ext.versions = [
         mifosdeposit            : '0.1.0-BUILD-SNAPSHOT',
         mifosteller             : '0.1.0-BUILD-SNAPSHOT',
         mifosreporting          : '0.1.0-BUILD-SNAPSHOT',
+        mifostemplate           : '0.1.0-BUILD-SNAPSHOT',
```
Comment (awasum) [inline build.gradle]: this has changed I think.

File: build.gradle [hunk 2]
```diff
@@ -85,7 +86,9 @@ dependencies {
             [group: 'io.mifos.portfolio', name: 'api', version: versions.mifosportfolio],
             [group: 'io.mifos.deposit-account-management', name: 'api', version: versions.mifosdeposit],
             [group: 'io.mifos.teller', name: 'api', version: versions.mifosteller],
-            [group: 'io.mifos.reporting', name: 'api', version: versions.mifosreporting],
+            //[group: 'io.mifos.reporting', name: 'api', version: versions.mifosreporting],
+
+            [group: 'io.mifos.template', name: 'api', version: versions.mifostemplate],
```
Comment (awasum) [inline build.gradle]: So has this...as u can see in sync gateway refactor.

File: src/main/java/io/mifos/dev/ServiceRunner.java [hunk 3]
```diff
@@ -218,13 +221,16 @@ public void before() throws Exception
     ServiceRunner.tellerManager = new Microservice<>(TellerManager.class, "teller", "0.1.0-BUILD-SNAPSHOT", ServiceRunner.INTEGRATION_TEST_ENVIRONMENT);
     startService(generalProperties, ServiceRunner.tellerManager);
 
-    ServiceRunner.reportManager = new Microservice<>(ReportManager.class, "reporting", "0.1.0-BUILD-SNAPSHOT", ServiceRunner.INTEGRATION_TEST_ENVIRONMENT);
-    startService(generalProperties, ServiceRunner.reportManager);
+    /*ServiceRunner.reportManager = new Microservice<>(ReportManager.class, "reporting", "0.1.0-BUILD-SNAPSHOT", ServiceRunner.INTEGRATION_TEST_ENVIRONMENT);
```
Comment (awasum) [inline src/main/java/io/mifos/dev/ServiceRunner.java]: Why is there no way to run 10 Microservices and their API managers at same time? What could be the problem eehh? is therea port restriction of restriction on number of services to run?

File: src/main/java/io/mifos/dev/ServiceRunner.java [hunk 4]
```diff
@@ -233,6 +239,7 @@ public void tearDown() throws Exception {
     ServiceRunner.customerManager.kill();
     ServiceRunner.organizationManager.kill();
     ServiceRunner.identityManager.kill();
+    ServiceRunner.templateManager.kill();
```
Comment (awasum) [inline src/main/java/io/mifos/dev/ServiceRunner.java]: Kill template/Sync first...as it was last start...FIFO I think.

File: src/main/java/io/mifos/dev/ServiceRunner.java (hunk 3 above)
Comment (Anh3h) [reply src/main/java/io/mifos/dev/ServiceRunner.java]: Space management

File: src/main/java/io/mifos/dev/ServiceRunner.java (hunk 4 above)
Comment (Anh3h) [reply src/main/java/io/mifos/dev/ServiceRunner.java]: Okay

**Comment by awasum** [comment]
Have u corrected this? @Anh3h ??

**Pull request merged**